src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java

```diff
@@ -696,6 +699,85 @@ public void writeChunkSection(ByteBuf buf, ChunkSection section) {
         this.writeDataPalette(buf, section.getBiomeData());
     }
 
+	public <E extends Enum<E>> EnumSet<E> readEnumSet(ByteBuf buf, Class<E> enumClass) {
+		E[] values = enumClass.getEnumConstants();
```

Comment:
Perhaps we can make this more performant if we instead reference `EnumClass.VALUES`?  `getEnumConstants` will clone the enum array. Ping me on Discord if this doesn't make sense.